invenio_rdm_records/services/community_records/service.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RDM Community Records Service."""
+from invenio_pidstore.errors import PIDDoesNotExistError
+from invenio_records_resources.services import (
+    LinksTemplate,
+    RecordService,
+    ServiceSchemaWrapper,
+)
+from invenio_records_resources.services.errors import PermissionDeniedError
+from invenio_records_resources.services.uow import unit_of_work
+from invenio_search.engine import dsl
+
+from invenio_rdm_records.proxies import current_record_communities_service
+
+
+class CommunityRecordsService(RecordService):
+    """Community records service.
+
+    The record communities service is in charge of managing the records of a given community.
+    """
+
+    @property
+    def community_record_schema(self):
+        """Returns the community schema instance."""
+        return ServiceSchemaWrapper(self, schema=self.config.community_record_schema)
+
+    @property
+    def community_cls(self):
+        """Factory for creating a community class."""
+        return self.config.community_cls
+
+    def read(
+        self, identity, community_id, params=None, search_preference=None, **kwargs
+    ):
+        """Search for records published in the given community."""
+        self.require_permission(identity, "read")
+        self.community_cls.pid.resolve(community_id)  # Ensure community's existence
+
+        params = params or {}
+
+        search_result = self._search(
+            "search",
+            identity,
+            params,
+            search_preference,
+            record_cls=self.record_cls,
+            search_opts=self.config.search,
+            extra_filter=dsl.Q("term", **{"parent.communities.ids": str(community_id)}),
+            permission_action="read",
+            **kwargs,
+        ).execute()
+
+        return self.result_list(
+            self,
+            identity,
+            search_result,
+            params,
+            links_tpl=LinksTemplate(
+                self.config.links_search_community_records,
+                context={
+                    "args": params,
+                    "id": community_id,
+                },
+            ),
+            links_item_tpl=self.links_item_tpl,
+        )
+
+    def _remove(self, community, record, identity):
+        """Remove a community from the record."""
+        data = dict(communities=[dict(id=str(community.id))])
+        errors = current_record_communities_service.delete(
+            identity, record_id=record.pid.pid_value, data=data
+        )
+
+        return errors
+
+    @unit_of_work()
+    def delete(self, identity, community_id, data, revision_id=None, uow=None):
+        """Remove records from a community."""
+        community = self.community_cls.pid.resolve(community_id)
+        self.require_permission(identity, "delete_record", record=community)
```

Comment:
I am wondering if we should use the term `remove` for the permission instead? It might be misleading. What do you think?
```suggestion
        self.require_permission(identity, "remove_record", record=community)
```

Reply:
Initailly I was using `remove` but @ppanero commented [here](https://github.com/inveniosoftware/invenio-rdm-records/pull/1178#discussion_r1104515132) that in geenral we do use delete for other resources. I agree that it's more clear to use remove (initially I was using `remove_record_from_community` to be more explicit) but I don't have a strong preference. Maybe others can share their opinions.